packages/redis/test/Adaptor.test.js

```diff
@@ -0,0 +1,334 @@
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('patient')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if key is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('', 'kwaku')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+});
+
+describe('hset', () => {
+  it('should set fields value for a specified key', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+
+    const state = {};
+    const result = await hset('patient', { name: 'kwaku' })(state);
+    expect(result.data).to.eql(1);
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+    const state = { key: 'patient', value: { name: 'mtuchi' } };
+    const result = await hset(
+      s => s.key,
+      s => s.value
+    )(state);
+    expect(result.data).to.eql(1);
+  });
+  it('should throw an error if key and value is not specified', async () => {
+    setMockClient({
+      hSet: async () => {
+        throw new Error(
+          'TypeError: Cannot convert undefined or null to object'
+        );
+      },
+    });
+
+    const state = {};
+    try {
+      await hset()(state);
+    } catch (error) {
+      expect(error.message).to.eql(
+        'TypeError: Cannot convert undefined or null to object'
+      );
+    }
+  });
+  it('should throw an error if value is not specified', async () => {
+    setMockClient({
+      hSet: async () => {
+        throw new Error(
+          'TypeError: Cannot convert undefined or null to object'
+        );
+      },
+    });
+
+    const state = {};
+    try {
+      await hset('patient')(state);
+    } catch (error) {
+      expect(error.message).to.eql(
+        'TypeError: Cannot convert undefined or null to object'
+      );
+    }
+  });
+  it('should set an empt key if key is not specified', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+
+    const state = {};
+    const result = await hset('', { name: 'kwaku' })(state);
+    expect(result.data).to.eql(1);
+  });
+});
+describe('scan', () => {
+  it('should return empty array if key not found', async () => {
+    setMockClient({
+      scan: async () => ({ keys: [], cursor: 0 }),
+    });
+    const state = {};
+    const result = await scan('*:abc*')(state);
+    expect(result.data).to.eql([]);
+    expect(result.cursor).to.eql(0);
+  });
+  it('should return collection of keys', async () => {
+    setMockClient({
+      scan: async () => ({ keys: ['noderedis:animals:1'], cursor: 0 }),
```

Comment:
Yes the `cursor` starts from `0` that's why `0` is the default

Reply:
`state.cursor` is the cursor at the end of the operation. Depending on the result that you get it can vary from 0 to cursor at the end of operation

Reply:
I would rather it had a value of `1` please, so that it is different to the default. Because at least that way it looks like the function has iterated over a collection, rather than just stayed at the first index.

I know it doesn't really matter but a cursor value of 0 just looks wrong to me, which reduces my confidence in the test.